test/Api.Test/Tools/Validators/SendRotationValidatorTests.cs

```diff
@@ -0,0 +1,130 @@
+﻿using Bit.Api.Tools.Models;
+using Bit.Api.Tools.Models.Request;
+using Bit.Api.Tools.Validators;
+using Bit.Core.Entities;
+using Bit.Core.Exceptions;
+using Bit.Core.Tools.Entities;
+using Bit.Core.Tools.Enums;
+using Bit.Core.Tools.Models.Data;
+using Bit.Core.Tools.Repositories;
+using Bit.Core.Tools.Services;
+using Bit.Test.Common.AutoFixture.Attributes;
+using NSubstitute;
+using Xunit;
+
+namespace Bit.Api.Test.Tools.Validators;
+
+[SutProviderCustomize]
+public class SendRotationValidatorTests
+{
+    [Fact]
+    public async Task ValidateAsync_Success()
+    {
+        // Arrange
+        var sendService = Substitute.For<ISendService>();
+        var sendRepository = Substitute.For<ISendRepository>();
+
+        var sut = new SendRotationValidator(
+            sendService,
+            sendRepository
+        );
+
+        var user = new User { Id = new Guid() };
+        var sends = CreateSends();
+        var userSends = sends.Select(s =>
+            new Send { Id = s.Id.Value, Type = s.Type, Data = "{}" }).ToList();
+        sendRepository.GetManyByUserIdAsync(user.Id).Returns(userSends);
+
+        // Act
+        var result = await sut.ValidateAsync(user, sends);
+
+        // Assert
+        Assert.Equal(userSends, result);
+    }
+
+    [Fact]
+    public async Task ValidateAsync_InputMissingUserSend_Throws()
+    {
+        // Arrange
+        var sendService = Substitute.For<ISendService>();
+        var sendRepository = Substitute.For<ISendRepository>();
+
+        var sut = new SendRotationValidator(
+            sendService,
+            sendRepository
+        );
+
+        var user = new User { Id = new Guid() };
+        var sends = CreateSends();
+        var userSends = sends.Select(s =>
+            new Send { Id = s.Id.Value, Type = s.Type, Data = "{}" }).ToList();
+        userSends.Add(new Send { Id = Guid.NewGuid(), Data = "{}" });
+        sendRepository.GetManyByUserIdAsync(user.Id).Returns(userSends);
```

Comment:
🎨 This could be more efficiently arranged by using an empty set and simply checking to see if any result causes ValidateAsync to throw. If at least one item is necessary, then just set up a single send to be returned. This also eliminates the need to perform computation that relies on the input data in the test.